geostreams/geodashboard
# Geostreams Component Library / Repository Split

## Description
This PR changes the gd-core, gd-core__old and gd-geostreaming into packages which are published on NPM registry. The rest of the folders are removed and they should live in their seperate repositories. The packages are inside the packages folder and are using Lerna for release and dependency management. Instructions for running and setting up have been added to the README file.

Overview of all other changes:
- Updated React to v17, Webpack to v5. Packages are still backwards compatible with react version 16. 
- Updated other dependencies(except material-ui-v5 and d3) and removed deprecated/unused modules.
- Added yarn work-spaces config
- Created an eslint-config module which can be imported and used by our projects. 

The template folder contains a template project which can be used to test the packages. 

### How to test:
Before testing, please make sure you have yarn installed. To install, run:  npm install --global yarn
#### Test whether the geostreaming and core apps work without a project:
  ```bash
 # Run following commands from the root folder
  yarn install
  yarn geostreaming # Runs the geostreaming app
  yarn core # Runs the core app
  ```
#### Test linking with project:
```bash
# From the repository root run
yarn link:all 
# From the project root (use template for testing)
yarn 
yarn link:geostreams
yarn start
```

The gltg and glm repositories are up to date as well to work with this setup. 

### Future work:
 - Github Actions for Deploying actions and updating package versions on release in project repositories. 
 - The packages are not currently bundled into commonjs or esm format. Hence, babel has to compile all these packages in the project. This has been handled in the template project folder(Refer to the babel config in webpack.common.js file). Having the packages bundled into one file creates issues with npm linking as after every change, `npm build` would have to be run for the changes to appear in the project. (This can be fixed with rollup but I have not been able to make it work properly yet)
 

## Review Time Estimate


## Types of changes (select all that applies)


## Checklist:


- [x] This change set requires updating dependencies.
